htdocs/societe/list.php

```diff
@@ -517,20 +519,20 @@
 	$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."societe_commerciaux as sc ON sc.fk_soc = s.rowid";
 	//elseif ($search_sale || (empty($user->rights->societe->client->voir) && (empty($conf->global->MAIN_USE_ADVANCED_PERMS) || empty($user->rights->societe->client->readallthirdparties_advance)) && !$socid)) $sql .= ", ".MAIN_DB_PREFIX."societe_commerciaux as sc";
 } elseif (!empty($search_sale) && $search_sale != '-1' || (empty($user->rights->societe->client->voir) && !$socid)) {
-	$sql .= ", ".MAIN_DB_PREFIX."societe_commerciaux as sc";
+	$sql .= " INNER JOIN ".MAIN_DB_PREFIX."societe_commerciaux as sc";
```

Comment:
The link to sale representative to manage external user must be rewritten. Currently there is some code into the SELECT and some into the WHERE.
Here you move everything into the FROM, but to be ready with the future advanced filter tool, we must evolve to have all the condition into the WHERE by using a WHERE EXISTS ...
Can you try this way. Should be: Nothing into the FROM
and into the WHERE
```
// Filter to retreive lines only if there is a filter on a sale representative (explicit filter or because external user)
if (((empty($user->rights->societe->client->voir) && !$socid) || ($search_sale && $search_sale != '-1')) {
    $sql .= " AND EXISTS (SELECT rowid FROM ".MAIN_DB_PREFIX."societe_commerciaux as sc WHERE sc.fk_soc = s.rowid";
    if (empty($user->rights->societe->client->voir) && !$socid) {
        $sql .= " AND sc.fk_user = ".((int) $user->id;        
    }
    if ($search_sale > 0) {
	    $sql .= " AND sc.fk_user = ".((int) $search_sale);
    }
    $sql .= ")";
}
// Filter to retreive lines only if there is no link with a sale representative
if ($search_sale && $search_sale == '-2') {
   $sql .= " AND NOT EXISTS (SELECT rowid FROM ".MAIN_DB_PREFIX."societe_commerciaux as sc WHERE sc.fk_soc = s.rowid)";
}
```

Reply:
hello
I understand the way but not sure they are more performant...